src/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/RC2Implementation.cs

```diff
@@ -0,0 +1,85 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Security.Cryptography;
+
+namespace Internal.Cryptography
+{
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Security", "CA5350")]
+    internal sealed partial class RC2Implementation : RC2
+    {
+        private const int BitsPerByte = 8;
+        private static readonly RandomNumberGenerator s_rng = RandomNumberGenerator.Create();
```

Comment:
I think just putting an internal static method\field on RandomNumberGenerator would work although we may need to be concerned about thread safety if used incorrectly. I think that could\should be addressed by a new issue however and not here. Thanks